### ISSUE TITLE
Help

### ISSUE THREAD

**Issue by ChilledDayz** [Mithion/ArsMagica2#1372]
I want to help test as I really like this mod, but I don't know how to compile the dev version. Any help would be very appreciated.

**Comment by Mithion** [comment]
Hey chilled,

Glad to hear you want to help test! :)  it's very much appreciated!

The steps to do so are pretty easy.  I'll do my best to be clear and to help you out.

1.) Download Eclipse Java EE from https://eclipse.org/downloads/

2.) Install GIT.  Git Client for Windows is the easiest one I find.

3.) Add AM2 as a repo.  The git client will have instructions on how to do this.  The repo URL is:
https://github.com/Mithion/ArsMagica2.git

4.) Pull the repo down.  The git client can do this with a single button once the repo is added.  All you need to do is tell it where you want it to put the files.

5.) Now we're actually at the mod itself.  Launch eclipse.  When asked for a project location, go to the folder that you specified for the repo download, and select the folder inside there named 'eclipse'.  

6.) Tell eclipse to run in debug mode (icon looks like a green bug).

7.) That's it!  You're debugging AM2.

**Comment by ChilledDayz** [comment]
Thanks for the fast response Mithion. I'll be glad to help, as AM2 has been my favourite magic mod for a while now. Those instructions are pretty clear hopefully I'll have no problems. I never could get my head around coding, it's why I stick to making game assets lol.

**Comment by ChilledDayz** [comment]
Having trouble with git for Windows. Been trying but can't work it out :-(

**Comment by Stalkstalks** [comment]
Mithion, I'm trying to build using your instructions, and I hit a snag at step 5, because there doesn't appear to be an "eclipse" folder in what git pulled down.

**Comment by Mithion** [comment]
Whoops, re-reading seems I skipped a step or three.  Sorry!
Do these before step 4 (if after you just need to relocate the git repo afterwards, not a big deal)

1.) You need to download Forge from files.minecraftforge.net.  Make sure you download forge for 1.7.10.  Extract/install it to the location of your choice.
2.) Open a command prompt to the install location and run the following commands:
    gradlew.bat setupDecompWorkspace
    gradlew.bat eclipse

After this you should see a src folder gets created, that's where you move the repo to.  You will also see an eclipse folder.  After that, continue with the steps above.

Again, sorry about the confusion.

**Comment by ChilledDayz** [comment]
Okay thanks I'll give it a try later

**Comment by ChilledDayz** [comment]
I think I'll just give up. i'm really not good at this stuff, i guess i'll just have to wait for a new update to test this out .thanks for trying to help me though i really do appreciate it. i don't want to waste anymore of your time

**Comment by Stalkstalks** [comment]
I've failed at getting eclipse to actually build the source and allow me to debug it, however, I have managed to get a jar to build! So at least I can fiddle around with the current dev version and find bugs in it, even if I can't fix them myself.

Mithion's modified instructions work, mostly. in 3.5.2, the command prompt has to be opened into the folder where you had git put the project. Run gradlew.bat with the two commands he gives, and then run
"gradlew.bat build" at the same command prompt, and it will build the project

I've dropped the resulting .jar into a minecraft instance (after renaming to mark it as a dev version built today) and it works!
Still need to figure out how to get it to build in eclipse so I can use its debugger, but that's a start, at least. Thanks for your help, Mithion!
